Add lp-ocp-compat-cr-onboarding agent skill (#45)
CSPI-QE/ocp-ci-docs: docs/OCP_CI_Tutorials/Reporting/Reporting_Guide.md

> **IMPORTANT**
>
> When defining the `FIREWATCH_CONFIG` variable, please try to cover every step that is executed during your scenario, you can view the steps that are run by going to a recent run of your scenario and viewing the artifacts. Each step should have a folder for it's artifacts and logs that you can use to build your config. If you happen to miss one of the steps and a failure occurs in that step, it will cause the failure to not match any of the rules in the config. In that case, a generic bug for the failure will be filed in the `FIREWATCH_DEFAULT_JIRA_PROJECT` project.

## Component Readiness

> This section outlines the action items required to onboard a layered product into the [Component Readiness](https://sippy.dptools.openshift.org/sippy-ng/component_readiness/main?view=4.22-main) tool.

Reviewer: Please remove the reference to a specific view, as it will become stale when the OCP release is stepped.
```suggestion
This section outlines the action items required to onboard a layered product into the [Component Readiness](https://sippy.dptools.openshift.org/sippy-ng/component_readiness/main) tool.
```

References:

> - Our `OCP 4.22` (current) Component Readiness view: [4.22-LP-Interop](https://sippy.dptools.openshift.org/sippy-ng/component_readiness/main?view=4.22-LP-Interop)

Reviewer: This increases the maintenance burden, as we would have to continuously update the current view. Please remove.

Instead, you should insert a reference to the GitHub repository where the changes should be made.

Author: Updated the whole section, please check commits diff.

- This tool is maintained by the SHIP and TRT teams, which are available in the `#forum-ocp-release-oversight` Slack channel.

For job-level CR compliance requirements in scenario configuration, see [Make a Job CR-Compliant](../Scenario_Development/Scenario_Development_Guide.md#make-a-job-cr-compliant) in the Scenario Development Guide.

### Sippy

This document is a **checklist for coding agents** (and humans) adding support in [Sippy](https://github.com/openshift/sippy) for a new **MY-CMP** product that publishes CI under the layered-product / lp-interop pattern (JUnit suite like `MY-CMP-lp-interop`, Prow jobs under `…-lp-interop-…`).

---

#### Prerequisites: gather required information

> From `openshift/release` (or your team’s CI config), confirm:

Reviewer: The term `team’s CI config` is confusing, especially in reality we work with other team Jobs. A good technical doc. should refrain using 1st & 2nd party designation.
```suggestion
Within the `openshift/release` repository, under CI Configuration files (`ci-operator/config/**/*.yaml`), confirm:
```

> 1. **Mapped `testSuites` component name** as it appears in imported test data (often `ProductName-lp-interop`, e.g. `OADP-lp-interop`). It must match **exactly** (case-sensitive) what you add to `testSuites`.

Reviewer: Be discipline on presenting pattern vs example.
```suggestion
1. **Mapped `testSuites` component name** as it appears in imported test data (often `<ProductName>-lp-interop`, e.g. `OADP-lp-interop`). It must match **exactly** (case-sensitive) what you add to `testSuites`.
```

Also, suddenly you refers to `testSuites` in here. Assuming someone does not knows anything about CR, do you think that person understand what need to be done after reading this doc.? I think it will be helpful if a link where the `testSuites` is defined.

Author: Addressed.

2. **Stable substring of periodic name**, e.g. `-lp-interop-cr- `. The variant registry matches **literal substrings** on the lowercased job name (first match wins).

>    - Whether you need **multiple** patterns (e.g. `-lp-interop-cr-acs` vs `-lp-interop-cr-acs-latest`)—add **separate** rows, **more specific before more general**.

Reviewer: Better wording:
```suggestion
  - If **multiple** patterns are required (e.g., `-lp-interop-cr-acs` and `-lp-interop-cr-acs-latest`), add **separate** rows, ensuring the **more specific patterns precede the more general ones**.
```

#### Note for AI / automation assistants (Sippy)

Do **not** run any `make` commands (or substitute commands) in this repository on behalf of the user. A maintainer must run them locally when noted below.

---

#### 1. Allow importing tests: `pkg/db/suites.go`

**File:** `pkg/db/suites.go`
**List:** `testSuites`

- Append the **exact** mapped JUnit suite name, e.g. `MyProduct-lp-interop`.
- Keep the list **sorted in a sensible way** (group with other `*-lp-interop` suites).

Tests from suites **not** in this list are **not** imported into Sippy’s DB.

---

#### 2. Map Prow job names → `LayeredProduct`: `pkg/variantregistry/ocp.go`

**File:** `pkg/variantregistry/ocp.go`
**Function:** `setLayeredProduct`

Add an entry to `layeredProductPatterns`:

```go
{"-lp-interop-cr-my-cmp", "lp-interop-my-cmp"},
```

**Rules:**

- **`product` value:** always use the **`lp-interop-…`** form (lowercase, hyphenated), e.g. `lp-interop-my-cmp`. This is what Component Readiness views filter on.
- **`substring`:** must appear in real periodic job names after lowercasing. Align with CI naming (often `-lp-interop-cr- `).

> - **Order matters:** the slice is scanned **top to bottom**; the **first** match wins. Place **narrow** patterns (e.g. product-specific) **above** broad patterns like `{"-virt", "virt"}` so lp-interop jobs are not misclassified.

Reviewer 2: Maybe also add this as its own Note or Warning block to ensure someone doesn't accidentally place an LP-specific check below a generic catch-all check in `ocp.go`

Author: See added section:
https://github.com/CSPI-QE/ocp-ci-docs/pull/45/changes#diff-880fd9db9bcd4a5b4bea893cb3a822c3ddcd8b9744b0d2ac7e9305789dbbe868R245-R247

```

which rewrites `pkg/variantregistry/snapshot.yaml`.

> Expect snapshot tests to fail until `make update-variants` has been run by a maintainer.

---

#### 6. What **not** to do

- Do **not** run **`make`** (including `make update-variants`, `make`, `make test`, etc.) from the agent; record **`make update-variants`** for the human when variant code changed.
- Do **not** hand-edit **`snapshot.yaml`** unless you have a documented, repo-approved process; prefer **`make update-variants`**.
- Do **not** change unrelated views, suites, or variant patterns.
- After frontend changes under `sippy-ng`, this onboarding path does not require npm; if you touch JS, follow `AGENTS.md` (eslint/prettier) separately.

---

#### Summary checklist

| Step | Location | Action |
|------|----------|--------|
| 1 | `pkg/db/suites.go` | Add JUnit suite `MyProduct-lp-interop` to `testSuites` |
| 2 | `pkg/variantregistry/ocp.go` | Add `setLayeredProduct` pattern → `lp-interop-my-cmp` |
| 3 | `config/views.yaml` | Add `lp-interop-my-cmp` to `*-LP-Interop` views’ `LayeredProduct` |
| 4 | `pkg/variantregistry/ocp_test.go` | Add `TestVariantSyncer` case (recommended) |

> | 5 | Maintainer | Run **`make update-variants`** after variant changes |

Reviewer: Please make this readable in the source code by aligning the `|` characters properly. Since we work at the source-code level, having the `|` aligned significantly improves readability.

Author: I didn't like the results of the indented table. Agent said that numbered lists are strong enough. Please compare to this commit: [d4ac3ee38f913961ad4942a307627377d688cc4e](https://github.com/CSPI-QE/ocp-ci-docs/pull/45/changes/d4ac3ee38f913961ad4942a307627377d688cc4e)

Reviewer: This is the result of the aligned (not indented) table, right:
<img width="1122" height="189" alt="image" src="https://github.com/user-attachments/assets/0264caa6-5154-417a-8963-2a26eff8a9e3" />

It looks great to me compare to:
<img width="764" height="145" alt="image" src="https://github.com/user-attachments/assets/9f9f45be-517a-4f3f-b8ee-032efea53386" />

But if you want to change to number list, it's fine too. My point is that if you use table, please make it aligned in the source code.

    [RedHatQE/OpenShift-LP-QE--Tools](https://github.com/RedHatQE/OpenShift-LP-QE--Tools)):

    ```bash
    #!/bin/bash

    set -o nounset
    set -o errexit

>     set -o pipefail

Reviewer 3: nit: Consider adding `set -o xtrace` and `shopt -s inherit_errexit` for completeness. `xtrace` gives you a command trace in CI logs when steps fail. `inherit_errexit` matters here specifically because the `eval "$(... curl/wget ...)"` below relies on command substitution; without it, a network failure inside `$(...)` is silently swallowed.

The SKILL.md in this same PR already uses the complete form (`set -euxo pipefail; shopt -s inherit_errexit`), so this would bring the Reporting Guide example into alignment.

```suggestion
    #!/bin/bash

    set -o nounset
    set -o errexit
    set -o xtrace
    set -o pipefail
    shopt -s inherit_errexit
```

Author: Ah, this is what happens when you ask ai to align it according to the best practices, and it doesn't do it properly :( Thanks for noticing.
